demos/shadow/shadow_demo_with_mbedtls/shadow_demo_with_mbedtls.c

```diff
@@ -0,0 +1,1115 @@
+    {
+        LogInfo( ( "Not find device private key and certificate in PKCS #11." ) );
+    }
+
+    return xResult;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of shadow demo.
+ *
+ * This main function demonstrates how to use the macros provided by the
+ * Device Shadow library to assemble strings for the MQTT topics defined
+ * by AWS IoT Device Shadow. Named shadow topic strings differ from unnamed
+ * ("Classic") topic strings as indicated by the tokens within square brackets.
+ *
+ * The main function uses these macros for topics to subscribe to:
+ * - SHADOW_TOPIC_STR_UPDATE_DELTA for "$aws/things/thingName/shadow[/name/shadowname]/update/delta"
+ * - SHADOW_TOPIC_STR_UPDATE_ACC for "$aws/things/thingName/shadow[/name/shadowname]/update/accepted"
+ * - SHADOW_TOPIC_STR_UPDATE_REJ for "$aws/things/thingName/shadow[/name/shadowname]/update/rejected"
+ *
+ * It also uses these macros for topics to publish to:
+ * - SHADOW_TOPIC_STR_DELETE for "$aws/things/thingName/shadow[/name/shadowname]/delete"
+ * - SHADOW_TOPIC_STR_UPDATE for "$aws/things/thingName/shadow[/name/shadowname]/update"
+ *
+ * The helper functions this demo uses for MQTT operations have internal
+ * loops to process incoming messages. Those are not the focus of this demo
+ * and therefore, are placed in a separate file shadow_demo_helpers.c.
+ */
+int main( int argc, char ** argv )
+{
+    int returnStatus = EXIT_SUCCESS;
+    int demoRunCount = 0;
+    bool connectionEstablished = false;
+
+    bool status = false;
+    CK_RV pkcs11ret = CKR_OK;
+
+    /* Buffer for holding received certificate until it is saved. */
+    char certificate[ CERT_BUFFER_LENGTH ];
+    size_t certificateLength;
+
+    /* Buffer for holding the certificate ID. */
+    char certificateId[ CERT_ID_BUFFER_LENGTH ];
+    size_t certificateIdLength;
+
+    /* Handle for pkcs#11. */
+    CK_SESSION_HANDLE p11Session;
+
+    /* A buffer containing the update document. It has static duration to prevent
+     * it from being placed on the call stack. */
+    static char updateDocument[ SHADOW_REPORTED_JSON_LENGTH + 1 ] = { 0 };
+
+    ( void ) argc;
+    ( void ) argv;
+
+    do
+    {
+        /* Initialize the buffer lengths to their max lengths. */
+        certificateLength = CERT_BUFFER_LENGTH;
+        certificateIdLength = CERT_ID_BUFFER_LENGTH;
+
+        /* Initialize the PKCS #11 module */
+        pkcs11ret = xInitializePkcs11Session( &p11Session );
+
+        if( pkcs11ret != CKR_OK )
+        {
+            LogError( ( "Failed to initialize PKCS #11." ) );
+            status = false;
+        }
+        else
+        {
+            status = checkCertAndPrivateKeyStored( p11Session );
+
+            if ( !status )
+            {
+                LogInfo( ( "Need to load claim credentials..." ) );
+
+                /* Insert the claim credentials into the PKCS #11 module */
+                status = loadClaimCredentials( p11Session,
```

Comment:
can we change the function name to loadCredentials()? I think this is not claim credentials, just a normal credentials created by one click. trying to decouple those terms out from fleet provisioning.